client/driver/docker.go

```diff
@@ -269,15 +269,15 @@ func (d *DockerDriver) createContainer(ctx *ExecContext, task *structs.Task, dri
 
 		containerToHostPortMap := make(map[string]int)
 		for _, port := range network.DynamicPorts {
-			var containerPort int
-			if len(driverConfig.PortMap) > 0 {
-				var ok bool
-				containerPort, ok = driverConfig.PortMap[0][port.Label]
-				if !ok {
-					containerPort = port.Value
+			// By default we will map the allocated port 1:1 to the container
+			containerPort := port.Value
+
+			// If the user has mapped a port using port_map we'll change it here
+			if len(driverConfig.PortMap) == 1 {
+				mapped, ok := driverConfig.PortMap[0][port.Label]
+				if ok {
+					containerPort = mapped
 				}
-			} else {
-				containerPort = port.Value
 			}
 
 			containerPortStr := docker.Port(strconv.Itoa(containerPort))
```

Comment:
This stuff should probably be extracted to share code between the reserved and dynamic